scripts/kconfig/kconfig.py

```diff
@@ -173,32 +176,38 @@ def promptless(sym):
     return not any(node.prompt for node in sym.nodes)
 
 
-def write_kconfig_filenames(paths, root_path, output_file_path):
-    # 'paths' is a list of paths. The list has duplicates and the
-    # paths are either absolute or relative to 'root_path'.
+def write_kconfig_filenames(kconf, kconfig_list_path):
+    # Writes a sorted list with the absolute paths of all parsed Kconfig files
+    # to 'kconfig_list_path'. The paths are realpath()'d, and duplicates are
+    # removed. This file is used by CMake to look for changed Kconfig files. It
+    # needs to be deterministic.
 
-    # We need to write this list, in a format that CMake can easily
-    # parse, to the output file at 'output_file_path'.
-
-    # The written list has sorted real (absolute) paths, and it does not have
-    # duplicates. The list is sorted to be deterministic. It is realpath()'d
-    # to ensure that different representations of the same path does not end
-    # up with two entries, as that could cause the build system to fail.
-
-    with open(output_file_path, 'w') as out:
-        for path in sorted({os.path.realpath(os.path.join(root_path, path))
-                            for path in paths}):
+    with open(kconfig_list_path, 'w') as out:
+        for path in sorted({os.path.realpath(os.path.join(kconf.srctree, path))
+                            for path in kconf.kconfig_filenames}):
```

Comment:
For future reference, the review goes faster if you separate changes and refactorings into different commits.

Reply:
Because then when looking at refactoring patches I can trivially verify that the changes have no semantic effects.

And when looking at actual changes I won't have to sift through the noise of unrelated refactoring.

Reply:
Yeah, did some drive-by-ing and got lazy. Will split better next time.